fhir-persistence-blob/src/main/java/com/ibm/fhir/persistence/blob/BlobStorePayload.java

```diff
@@ -45,20 +62,22 @@ public BlobStorePayload(int resourceTypeId, String logicalId, int version, Strin
      * @param client
      * @throws FHIRPersistenceException
      */
-    public Response<BlockBlobItem> run(BlobManagedContainer client) throws FHIRPersistenceException {
-        final StringBuilder blobNameBuilder = new StringBuilder();
-        blobNameBuilder.append(resourceTypeId);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(logicalId);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(version);
-        blobNameBuilder.append("/");
-        blobNameBuilder.append(resourcePayloadKey);
-        BlobClient bc = client.getClient().getBlobClient(blobNameBuilder.toString());
+    public CompletableFuture<PayloadPersistenceResult> run(BlobManagedContainer client) throws FHIRPersistenceException {
         
-        BlobParallelUploadOptions uploadOptions = new BlobParallelUploadOptions(BinaryData.fromBytes(ioStream.getRawBuffer()));
-        Response<BlockBlobItem> response = bc.uploadWithResponse(uploadOptions, client.getProperties().getTimeout(), null);
-        
-        return response;
+        if (ioStream.size() > MAX_APPEND_BLOB_BLOCK_SIZE) {
+            throw new FHIRPersistenceException("Resource payload size cannot exceed " + MAX_APPEND_BLOB_BLOCK_SIZE + " bytes");
```

Comment:
does this mean we can only support offload for resources <4 MiB?

Reply:
Correct. Note added to README

Reply:
ok.  so if the resource payload does exceed this, I presume we'll fail the transaction with a 500 error?
if we're treating this as a "documented limitation", maybe we'd want a 400 error instead with an OperationOutcome that has an Issue with a type either "too-costly" or "business-rule"?

Reply:
Now returns a 413 resource too large